exasol_transformers_extension/udfs/models/base_model_udf.py

```diff
@@ -199,36 +215,10 @@ def clear_device_memory(self):
         """
         Delete models and free device memory
         """
-        self.last_loaded_model = None
-        self.last_loaded_tokenizer = None
+        self.model_loader.last_loaded_model = None
+        self.model_loader.last_loaded_tokenizer = None
```

Comment:
I would add a `clear` method to the ModelLoader setting these two variables to None.
If `last_laded_model` and `last_loaded_tokenizer` still need to be visible I would make them properties.

BTW, if the `last_created_pipeline` keeps references to these objects they won't be garbage-collected. So you probably need to it None too.

Reply:
will move the clear method from base udf also